cpp/ql/test/library-tests/dataflow/fields/partial-definition-diff.ql

```diff
@@ -0,0 +1,52 @@
+/**
+ * @kind problem
+ */
+
+import cpp
+import semmle.code.cpp.ir.dataflow.DataFlow::DataFlow as IR
+import semmle.code.cpp.dataflow.DataFlow::DataFlow as AST
+
+newtype TNode =
+  TASTNode(AST::Node n) or
+  TIRNode(IR::Node n)
+
+class Node extends TNode {
```

Comment:
Can we add a `getLocation` to this class?

Reply:
Good point! I've added this now.